pkg/internal/ebpf/tracer_linux.go

```diff
@@ -39,13 +39,27 @@ type instrumenter struct {
 	modules   map[uint64]struct{}
 }
 
-func resolveInternalMaps(spec *ebpf.CollectionSpec) (*ebpf.CollectionOptions, error) {
+func roundToNearestMultiple(x, n uint32) uint32 {
+	return (x + n/2) / n * n
+}
+
+// RingBuf map types must be a multiple of os.Getpagesize()
+func adjustRingBufMaxEntries(m *ebpf.MapSpec) {
+	if m.Type == ebpf.RingBuf {
+		m.MaxEntries = roundToNearestMultiple(m.MaxEntries, uint32(os.Getpagesize()))
+	}
+}
+
+// sets up internal maps and ensures sane max entries values
+func resolveMaps(spec *ebpf.CollectionSpec) (*ebpf.CollectionOptions, error) {
 	collOpts := ebpf.CollectionOptions{MapReplacements: map[string]*ebpf.Map{}}
 
 	internalMapsMux.Lock()
 	defer internalMapsMux.Unlock()
 
 	for k, v := range spec.Maps {
+		adjustRingBufMaxEntries(v)
```

Comment:
Small nit, I'd expose the `if m.Type == ebpf.Ringbuf ` test in this loop here, or maybe better rename the function to say something like `alignMaxEntriesIfRingBuf`.